src/staking-router-modules/contracts/lido-locator/lido-locator.service.ts

```diff
@@ -1,19 +1,30 @@
 import { LidoLocator, LIDO_LOCATOR_CONTRACT_TOKEN } from '@lido-nestjs/contracts';
 import { Injectable, Inject, LoggerService } from '@nestjs/common';
-import { ExecutionProvider } from 'common/execution-provider';
 import { BlockTag } from '../interfaces';
 import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
+// TODO: maybe ../../../ shows us that we need to move execution-provider on level up
+import { ConfigService } from '../../../common/config';
 
 @Injectable()
 export class LidoLocatorService {
   constructor(
     @Inject(LIDO_LOCATOR_CONTRACT_TOKEN) protected readonly contract: LidoLocator,
-    protected readonly provider: ExecutionProvider,
     @Inject(LOGGER_PROVIDER) protected readonly loggerService: LoggerService,
+    protected readonly config: ConfigService,
   ) {}
 
+  private getContract() {
+    const lidoAddress = this.config.get('LIDO_LOCATOR_ADDRESS');
+
+    if (lidoAddress) {
+      return this.contract.attach(lidoAddress);
+    }
+
+    return this.contract;
+  }
+
   async getStakingRouter(blockTag: BlockTag) {
     this.loggerService.log('Contract locator address', this.contract.address);
-    return await this.contract.stakingRouter({ blockTag } as any);
+    return await this.getContract().stakingRouter({ blockTag } as any);
```

Comment:
First, we log `this.contract.address` and than call `this.getContract()`. There might be a different address